### PR TITLE
Update JSON_FLAGS.md

### PR DESCRIPTION
**Pull request by RedMisao**
#### Summary
None

#### Purpose of change
1. Documented the `LEVITATE`, `MAGIC_FOCUS`, `NO_SPELLCASTING`, `SUBTLE_SPELL` and `SILENT_SPELL` flags
2. Removed "Ammo type" subsection from Ammo (not flags)
3. Removed "Covers" and "Specifically Covers" subsection from Armor (not flags, already in JSON_INFO.md)
4. Removed "Techniques" (not flags, is just a summary naming the .json files)
5. Removed the duplicate bionic-related flags from "Generic"
6. Removed the obsolete flags from "Death function"
7. Reorganized the doc
8. Updated the TODO
9. Checked overall style and grammar

### CODE REVIEW

**Comment by MNG-cataclysm** [comment]
“Documentation” isn’t a valid changelog ID. I’d replace it with something like “Infastructure”, or replace the entire line with “None”.

File: doc/JSON_FLAGS.md [hunk 1]
```diff
@@ -4,167 +4,398 @@
...
-- ```thrown``` Thrown
-- ```unfinished_char``` Semi-charred fuel
-- ```water``` Water
-- ```paper``` Paper
-
-### Ammo effects
+- Some ammo effects (`LASER`, `NEVER_MISFIRES`) can be set as ammo `"effects"`, ammo `"flags"` or as the gun's `"ammo_effects"`.  This could use some clarification.
+- Move the Use action subsection from Comestibles to JSON_INFO.md, as those are not flags.
+- Move the Parameters sub-subsection from Faults to JSON_INFO.md, as those are not flags.
+- Move the Categories subsection from Recipes to JSON_INFO.md, as those are not flags.
```
Comment (Fris0uman) [inline doc/JSON_FLAGS.md:86]: I don't think the documentaiton is the place to leave TODOs like this

File: doc/JSON_FLAGS.md [hunk 2]
```diff
@@ -1052,36 +1206,10 @@ Monster physical sizes.
...
-- ```MUTCAT_INSECT``` "You hear buzzing, and feel your body harden."
-- ```MUTCAT_LIZARD``` "For a heartbeat, your body cools down."
-- ```MUTCAT_MEDICAL``` "Your can feel the blood rushing through your veins and a strange, medicated feeling washes over your senses."
-- ```MUTCAT_PLANT``` "You feel much closer to nature."
-- ```MUTCAT_RAPTOR``` "Mmm...sweet bloody flavor...tastes like victory."
-- ```MUTCAT_RAT``` "You feel a momentary nausea."
-- ```MUTCAT_SLIME``` "Your body loses all rigidity for a moment."
-- ```MUTCAT_SPIDER``` "You feel insidious."
-- ```MUTCAT_TROGLOBITE``` "You yearn for a cool, dark place to hide."
+Special attacks have been moved to [MONSTER_SPECIAL_ATTACKS.md](MONSTER_SPECIAL_ATTACKS.md) as they have all been migrated away from flags.
```
Comment (Fris0uman) [inline doc/JSON_FLAGS.md:1212]: What's the point of this note though? If the information is in the other doc just remove it from here. I don't think we need to leave breadcrumbs around like this.

**Review by Fris0uman — COMMENTED** [review]
This is a lot of change in one go and it makes it pretty difficult to review

File: doc/JSON_FLAGS.md (hunk 2 above)
Comment (RedMisao) [reply doc/JSON_FLAGS.md:1212]: That sentence is from the old doc, I can remove the section entirely.

File: doc/JSON_FLAGS.md (hunk 1 above)
Comment (RedMisao) [reply doc/JSON_FLAGS.md:86]: Some old docs have TODO sections. In the case of the current JSON_FLAGS.md, it's asking for a review or the removal of the `Ammo type` table, which I removed in the PR as ammo types are fully .jsonized (and thus aren't flags).

However, I didn't consider correct to remove the three mentioned subsections right away, as those contain information that belong to a different doc, and may require a different PR for that doc altogether.

File: doc/JSON_FLAGS.md (hunk 2 above)
Comment (Fris0uman) [reply doc/JSON_FLAGS.md:1212]: Yeah just remove it

Comment (RedMisao) [reply doc/JSON_FLAGS.md]: Done, it's removed.

**Comment by Fris0uman** [comment]
Just going from the PR description, are the thing you removed documented somewhere else?

**Comment by RedMisao** [comment]
> Just going from the PR description, are the thing you removed documented somewhere else?

Only the Ammo section (including Ammo type subsection) was fully removed.
The large red chunks were ordered alphabetically or moved to different sections or subsections (i.e. armor layers from Armor to Armor -> Layers).

**Comment by Maleclypse** [comment]
Are you up for rebasing this to resolve conflicts?

**Pull request closed without merging**